compiler/z/codegen/ControlFlowEvaluator.cpp

```diff
@@ -338,21 +338,38 @@ xmaxxminHelper(TR::Node* node, TR::CodeGenerator* cg, TR::InstOpCode::Mnemonic c
    generateS390LabelInstruction(cg, TR::InstOpCode::label, node, cFlowRegionStart);
    cFlowRegionStart->setStartInternalControlFlow();
 
+   //Checking common Condition Code and branching to cFlowRegionEnd
    generateRREInstruction(cg, compareRROp, node, lhsReg, rhsReg);
    generateS390BranchInstruction(cg, TR::InstOpCode::BRC, branchCond, node, cFlowRegionEnd);
 
+   //Check for NaN operands for float and double
+   if (node->getOpCode().isDouble() || node->getOpCode().isFloat())
+      {
+      if (node->getOpCode().isDouble())
+         {
+         // If first operand is NaN, then we are done, otherwise fallthrough to move second operand as result
+         generateRREInstruction(cg, TR::InstOpCode::LTDBR, node, lhsReg, lhsReg);
+         generateS390BranchInstruction(cg, TR::InstOpCode::BRC, TR::InstOpCode::COND_CC3, node, cFlowRegionEnd);
+         }
+      else if (node->getOpCode().isFloat())
+         {
+         // If first operand is NaN, then we are done, otherwise fallthrough to move second operand as result
+         generateRREInstruction(cg, TR::InstOpCode::LTEBR, node, lhsReg, lhsReg);
+         generateS390BranchInstruction(cg, TR::InstOpCode::BRC, TR::InstOpCode::COND_CC3, node, cFlowRegionEnd);
+         }
+      }
```

Comment:
I think you can simplify the code above like following. 
```suggestion
   if (node->getOpCode().isFloatingPoint())
      {
      // If first operand is NaN, then we are done, otherwise fallthrough to move second operand as result
      generateRREInstruction(cg, node->getOpCode().isDouble() ? TR::InstOpCode::LTDBR : TR::InstOpCode::LTEBR , node, lhsReg, lhsReg);
      generateS390BranchInstruction(cg, TR::InstOpCode::BRC, TR::InstOpCode::COND_CC3, node, cFlowRegionEnd);
      }
```